ffi/dotnet-idsdk/IoticsIdLib/src/Tools.cs

```diff
@@ -0,0 +1,116 @@
+namespace IOTICS;
+
+class Tools {
+    
+    
+    public static string InvokeGoFunction(Func<IdLib.Return> goFunction)
+    {
+        string? error;
+        IdLib.Return result;
+
+        try
+        {
+            // Call the Go function
+            result = goFunction.Invoke();
+        }
+        catch (Exception ex)
+        {
+            // Handle any unexpected exceptions
+            throw new InvalidOperationException($"Exception when invoking method: {ex.Message}");
```

Comment:
throw with inner exception instead of just message?

Reply:
done